src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -945,4 +963,48 @@ private function getModelName(): string
         }
         return ClassInfo::shortName($this->record);
     }
+
+    /**
+     * Get Pjax response negotiator so form submission mirrors other form submission in the CMS.
+     * See LeftAndMain::getResponseNegotiator()
+     */
+    private function getResponseNegotiator(DBHTMLText $renderedForm): PjaxResponseNegotiator
+    {
+        return new PjaxResponseNegotiator([
+            'Default' => function () use ($renderedForm) {
+                return $renderedForm;
+            },
+            'Content' => function () use ($renderedForm) {
+                return $renderedForm;
+            },
+            'CurrentForm' => function () use ($renderedForm) {
+                return $renderedForm;
+            },
+            'Breadcrumbs' => function () {
+                return $this->renderWith([
+                    'type' => 'Includes',
+                    'SilverStripe\\Admin\\CMSBreadcrumbs'
+                ]);
+            },
+            'ValidationResult' => function () {
+                // Assume valid by default, mirroring LeftAndMain's response negotiator
+                return $this->prepareDataForPjax([
+                    'isValid' => true,
+                    'messages' => '',
+                ]);
+            }
+        ], $this->getToplevelController()->getResponse());
+    }
+
+    /**
+     * Convert an array of data to JSON and wrap it in an HTML tag as pjax is used and jQuery will parse this
+     * as an element on the client side in LeftAndMain.js handleAjaxResponse()
+     * The attribute type="application/json" denotes this is a data block and won't be processed by a browser
+     * https://html.spec.whatwg.org/#the-script-element
+     * See LeftAndMain::prepareDataForPjax()
+     */
+    private function prepareDataForPjax(array $data): string
```

Comment:
Code duplication with LeftAndMain::prepareDataforPjax() - it would make sense to make this a protected method on RequestHandler and remove it from LeftAndMain and here

Reply:
Fair point, I've moved it - there's another PR now for admin where it gets removed from `LeftAndMain`.

Reply:
OK - since the linked sink PR on the parent issue is currently red, it probably makes sense to redo the sink PR with the new admin PR included

Reply:
That failure is expected - there's a note about it right next to the link to the CI run. But for the sake of avoiding ping pong I'll update the CI run to include the new admin PR.